packages/jest/test/__snapshots__/react-enzyme.test.js.snap

```diff
@@ -14,6 +14,47 @@ exports[`enzyme mount basic 1`] = `
 </Greeting>
 `;
 
+exports[`enzyme mount conditional styles 1`] = `
+.emotion-0 {
+  background-color: black;
+}
+
+.emotion-11rveqsm-Greeting {
+  background-color: red;
+}
```

Comment:
I'm struggling with getting this to work right.  When rendering with conditional styles, it is inserting styles from other tests in the spec and I'm not entirely sure why.  I tried just filtering out the falsy values from the css prop first but that didn't help, so I must be doing something wrong.  Help appreciated.

Reply:
Interesting - I might be wrong as I'm just trying to guess what might be the culprit here but my guess is that for some reason one of those return too much:
https://github.com/emotion-js/emotion/blob/fa1c25dae5967e71bc28ee7828ecf6a9d84f09e3/packages/jest/src/create-serializer.js#L222-L223

So to fix this I would inspect what happens in those 2 calls.

Reply:
Okay, so I fixed this issue and only the styles for the element are being returned.  Yay!  I think there are two remaining issues:

1. `isShallowEnzymeElement` does not correctly identify shallow vs mounted elements who had a css prop that was fasly.  This results in the following when mounting

```html
<div
  className="emotion-0"
>
  <span
    css="unknown styles"
  >
    <span
      className="emotion-1"
    >
      Test content
    </span>
  </span>
</div>
```

2. Conditional styles are getting `css="unknown styles"` since the class name is not added so when the css prop is attempted to be cleaned up it isn't since `hasKnownClass` is false.

Any thoughts you might have on these issues would be greatly appreciated.

Reply:
Could you add a failing test? I have recognized some other problems with shallowly rendered components which I'm investigating right now~ so this would be helpful for me.